dispatch/apps/api/views.py

```diff
@@ -407,6 +407,42 @@ def articles(self, request, pk=None):
         view = ArticleViewSet.as_view({'get': 'topic'})
         return view(request, pk=pk)
 
+class FileViewSet(viewsets.ModelViewSet):
+    """
+    Viewset for File model views.
+    """
+    model = File
+    serializer_class = FileSerializer
+
+    def get_queryset(self):
+        queryset = File.objects.all()
+        q = self.request.query_params.get('q', None)
+        if q is not None:
+            # If a search term (q) is present, filter queryset by term against `name`
+            queryset = queryset.filter(name__icontains=q)
+        return queryset
+
+    def bulk_delete(self, request):
+        deleted = []
+        ids = self.request.data.get('ids', None)
```

Comment:
> FileField doesn't actually delete the file itself anyway

This is an even trickier problem because Django allows you to specify custom storage handlers. For example, ubyssey.ca uses Amazon S3 via the `django-storages` package, which handles file deletion on its own (quite well, actually).